### DISCUSSION TITLE
Static Files (.js and .css) Not Loading

### DISCUSSION THREAD

**Discussion by walidmujahid** [timlegge/docker-foswiki#53] [General]
I'm trying to get FOSWiki up and running on a Digitalocean Ubuntu 24.04 droplet. I tried both building the image and using the one from Docker Hub. I was just keeping it simple at first and running the `docker run -idt -p 80:80 timlegge/docker-foswiki`. 

The browser console shows ERR_CONNECTION_REFUSED errors for all .js and .css files, even those that should be served directly by nginx. Here's an example:

```
GET https://167.71.185.130/pub/System/PageOptimizerPlugin/cache/5e8727b206ad93db4e301dfe8d11be97.css net::ERR_CONNECTION_REFUSED
... many more similar errors ...
```

The nginx error log:
```
2024/08/30 21:20:41 [notice] 24#24: *1 ".*" matches "/", client: {CENSORING_MY_IP}, server: _, request: "GET / HTTP/1.1", host: "167.71.185.130"
2024/08/30 21:20:41 [notice] 24#24: *1 rewritten data: "/bin/view", args: "", client: {CENSORING_MY_IP}, server: _, request: "GET / HTTP/1.1", host: "167.71.185.130"
2024/08/30 21:20:43 [error] 24#24: *1 FastCGI sent in stderr: "WARNING - /var/www/foswiki/data/.htpasswd DOES NOT EXIST
WARNING - /var/www/foswiki/data/.htpasswd DOES NOT EXIST
WARNING - /var/www/foswiki/data/.htpasswd DOES NOT EXIST
WARNING - /var/www/foswiki/data/.htpasswd DOES NOT EXIST
WARNING - /var/www/foswiki/data/.htpasswd DOES NOT EXIST
WARNING - /var/www/foswiki/data/.htpasswd DOES NOT EXIST
WARNING - /var/www/foswiki/data/.htpasswd DOES NOT EXIST
WARNING - /var/www/foswiki/data/.htpasswd DOES NOT EXIST
WARNING - /var/www/foswiki/data/.htpasswd DOES N" while reading response header from upstream, client: {CENSORING_MY_IP}, server: _, request: "GET / HTTP/1.1", upstream: "fastcgi://127.0.0.1:9000", host: "167.71.185.130"
2024/08/30 21:20:43 [error] 24#24: *1 FastCGI sent in stderr: "OT EXIST
WARNING - /var/www/foswiki/data/.htpasswd DOES NOT EXIST
WARNING - /var/www/foswiki/data/.htpasswd DOES NOT EXIST
WARNING - /var/www/foswiki/data/.htpasswd DOES NOT EXIST
woops file /var/www/foswiki/pub/System/ImagePlugin/style.css does not exist for url /pub/System/ImagePlugin/style.css" while reading upstream, client: {CENSORING_MY_IP}, server: _, request: "GET / HTTP/1.1", upstream: "fastcgi://127.0.0.1:9000", host: "167.71.185.130"
```

**Comment by timlegge** [comment]
I hope to look at it this weekend.  Do you have any instructions on reproducing your setup?

**Reply by walidmujahid** [reply]
I installed the latest docker engine on the newly spun up Digitalocean droplet following: https://docs.docker.com/engine/install/ubuntu/#install-using-the-repository

Then I would just start the image with `docker run -idt -p 80:80 timlegge/docker-foswiki`, no changes made. 

Edit: sorry, as I install via apt, I meant to link to those instructions.

**Comment by timlegge** [comment]
Hi

You started the container on port 80/tcp (http) but you are trying to connect on 443/tcp (https).  I always use docker-compose to start the containers.  See https://github.com/timlegge/docker-foswiki/blob/master/docker-compose.2-simple-https.yml for a sample.

`
docker-compose -f docker-compose.2-simple-https.yml up
`

The other option is to:

docker run -idt -p 80:80 -p 443:443 timlegge/docker-foswiki

but you will need to configure the https manually.  You can also:

`docker exec -it [container_name] /bin/bash` 

and grep for the https in the config to see if you changed something to https:

`
grep https /var/www/foswiki/lib/LocalSite.cfg

`
```
wget https://[removed your IP]/pub/System/PageOptimizerPlugin/cache/5e8727b206ad93db4e301dfe8d11be97.css
--2024-08-31 18:53:28--  https://[removed your IP]/pub/System/PageOptimizerPlugin/cache/5e8727b206ad93db4e301dfe8d11be97.css
Connecting to [removed your IP]:443... failed: Connection refused.

wget http://[removed your IP]/pub/System/PageOptimizerPlugin/cache/5e8727b206ad93db4e301dfe8d11be97.css
--2024-08-31 18:53:35--  http://[removed your IP]/pub/System/PageOptimizerPlugin/cache/5e8727b206ad93db4e301dfe8d11be97.css
Connecting to [removed your IP]:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: 436997 (427K) [text/css]
Saving to: ‘5e8727b206ad93db4e301dfe8d11be97.css’

5e8727b206ad93db4e301dfe8d11be97.cs 100%[================================================================>] 426.75K  2.23MB/s    in 0.2s    

2024-08-31 18:53:36 (2.23 MB/s) - ‘5e8727b206ad93db4e301dfe8d11be97.css’ saved [436997/436997]

```

**Reply by walidmujahid** [reply]
I got working. I used the docker compose, replaced the key and certificate. I just had to make sure to add https when loading port 8443 in the browser.

**Comment by timlegge** [comment]
Glad to hear - docker compose makes it a little easier for sure.  I probably should update the documentation since the docker run is pretty much where you start...

Close the discussion if you are fine
